src/wrapper/qr_mumps_common.jl

```diff
@@ -0,0 +1,65 @@
+struct qrm_spmat{T}  # T ∈ Union{Complex{Float32}, Complex{Float64}, Float32, Float64}
+  irn :: Ptr{Cint}
+  jcn :: Ptr{Cint}
+  val :: Ptr{T}
+  m   :: Cint
+  n   :: Cint
+  nz  :: Cint
+  sym :: Cint
+  h   :: Ptr{Cvoid}
+end
+
+struct qrm_spfct  # HELP 
+  cperm_in :: Ptr{Cint}
+  icntl    :: NTuple{20, Cint}
+  rcntl    :: NTuple{10, Cdouble}
+  gstats   :: NTuple{10, Clong}
+  h        :: Ptr{Cvoid}
+end
```

Comment:
`qrm_spfct` is a decomposition of a `qrm_spmat` but we only have the pointer to a model `h` that stores the decomposition. We don't know if the factorization was done in single or double precision for example are we need it to call the specialized routines.

Reply:
I do not have any better idea than adding a dummy field  to the qrm_spfct type.

Reply:
We can parametrize `qrm_spfct` even if the attributes have fixed types.
I didn't know that.

```julia
mutable struct bob{T}
       a :: Float64
       b :: Int64
end

bob{Float32}(2.0, 3)
bob{ComplexF64}(3.0, 5)
```